src/common/buf.c

```diff
@@ -33,6 +33,13 @@ buf_expand_shell_variables(struct buf *s)
 				++p;
 			}
 			*p = '\0';
+
+			/* Don't expand single $ (as used by subshells) */
+			if (!strlen(environment_variable.buf)) {
```

Comment:
Could just do !foo.len here

Reply:
will change

Reply:
Hm.. are you sure about that?
It seems we are adding the remaining string to `environment_variable` and then just throw a `\0` in there.
Thus I think `.len` is actually out of sync with `strlen()` here.

Reply:
Whoops. Yes, sorry. Ignore my comment.